glide-core/redis-rs/redis/tests/test_cluster_async.rs

```diff
@@ -2,6 +2,92 @@
 #![cfg(feature = "cluster-async")]
 mod support;
 
+use std::cell::Cell;
+use tokio::sync::Mutex;
+
+use lazy_static::lazy_static;
+
+lazy_static! {
+    static ref CLUSTER_VERSION: Mutex<Cell<usize>> = Mutex::<Cell<usize>>::default();
+}
+
+/// Check if the current cluster version is less than `min_version`.
+/// At first, the func check for the Valkey version and if none exists, then the Redis version is checked.
+async fn engine_version_less_than(min_version: &str) -> bool {
+    let test_version = crate::get_cluster_version().await;
+    let min_version_usize = crate::version_to_usize(min_version).unwrap();
+    if test_version < min_version_usize {
+        println!(
+            "The engine version is {:?}, which is lower than {:?}",
+            test_version, min_version
+        );
+        return true;
+    }
+    return false;
+}
+
+/// static fuction to get the engin version. when version is look like 8.0.0 -> 80000 and 12.0.1 -> 120001
+async fn get_cluster_version() -> usize {
+    let cluster_version = CLUSTER_VERSION.lock().await;
+    if cluster_version.get() == 0 {
+        let cluster = crate::support::TestClusterContext::new(3, 0);
+
+        let mut connection = cluster.async_connection(None).await;
+
+        let cmd = redis::cmd("INFO");
+        let info = connection
+            .route_command(
+                &cmd,
+                redis::cluster_routing::RoutingInfo::SingleNode(
+                    redis::cluster_routing::SingleNodeRoutingInfo::Random,
+                ),
+            )
+            .await
+            .unwrap();
+
+        let info_result = redis::from_owned_redis_value::<String>(info).unwrap();
+
+        match parse_version_from_info(info_result.clone()) {
+            Some(version) => cluster_version.set(version),
+            None => println!("Invalid version string in INFO : {}", info_result),
+        }
```

Comment:
If we failed to get the version from the INFO command, is there a point of continuing?

```rust
cluster_version
  .set(
    parse_version_from_info(info_result.clone())
      .expect(
        format!("Invalid version string in INFO : {}", info_result)
      )
  );
```